File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
  let sum = 0;
  
  const dfs = (node, num = '') => {
    if (!node) return;
    
    num += node.val;
    if (!node.left && !node.right) sum += +num;
    
    dfs(node.left, num);
    dfs(node.right, num);
  }
  
  dfs(root);
  
  return sum;
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        def dfs(node, num = 0):
            nonlocal ans
            if not node:
                return
            num = num * 10 + node.val
            if not node.left and not node.right:
                ans += num
            dfs(node.left, num)
            dfs(node.right, num)
            
        ans = 0
        dfs(root)
        return ans
            